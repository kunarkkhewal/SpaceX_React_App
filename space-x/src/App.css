.App {
  text-align: center;
  background-color: #f2f2f2;
  font-size: calc(10px + 2vmin);
  color: black;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.App-header p {
  margin: 0;
  padding: 15px;
  font-weight: bold;
}

.App-main {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
}

.App-footer {
  padding: 15px;
}

