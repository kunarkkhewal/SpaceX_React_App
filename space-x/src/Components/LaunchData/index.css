.LaunchData-div {
    display: flex;
    flex-wrap: wrap;
}

.mission-data-div {
    background-color: white;
    border-radius: 10px;
    display : inline-block;
    box-sizing: border-box;
    margin : 10px 15px;
    padding : 15px;
    width: 100%;
}

.mission-data-div img {
    background-color: #f2f2f2;
}