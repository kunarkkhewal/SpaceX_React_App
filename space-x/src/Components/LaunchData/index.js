import React, {useState, useEffect} from 'react';
import axios from 'axios';
import './index.css'

const LaunchData = () => {
    const [launchData, setLaunchData] = useState([])

    const getLaunchData = async () => {
        await axios.get('https://api.spacexdata.com/v3/launches?limit=100')
        .then(data => {
            setLaunchData(data.data);
        })
        .catch(error => {
            console.log(" -------- NO DATA!!! => error => ", error)
        })
    }
    
    useEffect(() => {
        getLaunchData();
    })

    return(
        <div className='LaunchData-div'>
            {launchData.map((launch, key) => {
                const {
                    mission_name, 
                    launch_year, 
                    mission_id, 
                    launch_success, 
                    links:{mission_patch, mission_patch_small}
                } = launch
                return <div className='mission-data-div'>
                        <img src={mission_patch_small} alt="user"/>
                        <h3>{mission_name}</h3>
                        <p>Mission Ids:</p>
                        <ul>{!!mission_id.length && mission_id.map(id=>{
                            return <li>{id}</li>
                        })}</ul>
                        <p>Launch Year: {launch_year}</p>
                        <p>Successful Launch: {launch_success}</p>
                    </div>
            })}
        </div>
    )
} 

export default LaunchData;