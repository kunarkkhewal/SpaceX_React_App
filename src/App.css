.App {
  text-align: center;
  background-color: #f2f2f2;
  font-size: calc(10px + 2vmin);
  color: black;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.App-header p {
  margin: 0;
  padding: 15px;
  font-weight: bold;
}

.App-main {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
}

.App-footer {
  padding: 15px;
}

@media only screen and (min-width: 701px) {
  .App-main {
    display: flex;
    flex-direction: row;
    justify-content: space-around;
  }
}

@media only screen and (min-width: 1441px) {
  .App-main {
    max-width: 1441px;
  }
}


