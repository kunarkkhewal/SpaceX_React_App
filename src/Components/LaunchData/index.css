.LaunchData-div {
    display: flex;
    flex-wrap: wrap;
}

.mission-data-div {
    background-color: white;
    border-radius: 10px;
    display : inline-block;
    box-sizing: border-box;
    margin : 10px 15px;
    padding : 15px;
    width: 100%;
}

.mission-data-div img {
    background-color: #f2f2f2;
}

.mission-data-div ul li {
    list-style: none;
}

@media only screen and (min-width: 701px) {
    .LaunchData-div {
        width: 65%;
        font-size: 1rem;
    }

    .mission-data-div {
        box-sizing: border-box;
        margin : 10px 15px;
        padding : 15px;
        width: 40%;
        min-height: 50px;
        overflow: hidden;
    }

    .mission-data-div img {
        background-color: #f2f2f2;
        width: 100%;
    }
}

@media only screen and (min-width: 1025px) {
    .LaunchData-div {
        width: 70%;
    }

    .mission-data-div {
        box-sizing: border-box;
        width: 20%;
    }

    .mission-data-div img {
        background-color: #f2f2f2;
        width: 100%;
    }
}

@media only screen and (min-width: 1441px) {
    .LaunchData-div {
        margin: 0 2%;
    }
}