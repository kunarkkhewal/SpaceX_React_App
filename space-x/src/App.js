import React from 'react';
import './App.css';
import Filter from './Components/Filter';
import LaunchData from './Components/LaunchData';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <p>
          SpaceX Launch Programs
        </p>
      </header>

      <main className="App-main">
        {<Filter/>}
        {<LaunchData/>}
      </main>


      <footer className="App-footer">
        Developed By: Kunark Khewal
      </footer>
    </div>
  );
}

export default App;
