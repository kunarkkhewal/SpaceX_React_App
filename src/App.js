import React, {useState} from 'react';
import './App.css';
import Filter from './Components/Filter';
import LaunchData from './Components/LaunchData';

export const DataContext = React.createContext();

function App() {
  const [missionData, setMissionData] = useState([]);

  return (
    <div className="App">
      <header className="App-header">
        <p>
          SpaceX Launch Programs
        </p>
      </header>

      <main className="App-main">
        <DataContext.Provider value={{missionData,setMissionData}}>
          {<Filter/>}
          {<LaunchData/>}
        </DataContext.Provider>
      </main>


      <footer className="App-footer">
        Developed By: Kunark Khewal
      </footer>
    </div>
  );
}

export default App;
