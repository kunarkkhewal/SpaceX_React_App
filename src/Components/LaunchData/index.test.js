import React from "react";
import { mount } from "enzyme";
import LaunchData from "./index";
import {DataContext} from './../../App';

describe("LaunchData", () => {
    it('should render LaunchData component correctly', () => {
        const component = mount(<LaunchData/>, {
            wrappingComponent: DataContext.Provider,
            wrappingComponentProps: { value: { missionData: [{
                mission_name:'mock-name', 
                launch_year: 2006, 
                mission_id: 1, 
                launch_success: false, 
                links:{mission_patch_small: 'mock-img'},
                rocket: {first_stage: {cores: [{land_success: false}]}}
            }] } }
        });

        expect(component).toMatchSnapshot();
    });
});