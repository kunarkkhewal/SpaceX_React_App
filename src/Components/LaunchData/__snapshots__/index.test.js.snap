// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LaunchData should render LaunchData component correctly 1`] = `
<LaunchData>
  <div
    className="LaunchData-div"
  >
    <div
      className="mission-data-div"
    >
      <img
        alt="user"
        src="mock-img"
      />
      <h3>
        mock-name
      </h3>
      <p>
        <b>
          Mission Ids:
        </b>
      </p>
      <ul />
      <p>
        <b>
          Launch Year:
        </b>
         
        2006
      </p>
      <p>
        <b>
          Successful Launch:
        </b>
         
        <span>
          false
        </span>
      </p>
      <p>
        <b>
          Successful Landing:
        </b>
         
        <span>
          false
        </span>
      </p>
    </div>
  </div>
</LaunchData>
`;
